content/automate/ManageIQ/System/CommonMethods/MiqAe.class/__methods__/weightedupdatestatus.rb

```diff
@@ -122,6 +136,11 @@ def main
                   # We set the task message.
                   if @handle.root['ae_state_step'] == 'on_error'
                     task.message = 'Failed'
+                    create_cleanup_request(task)
+                  elsif task.get_option('cancel_requested')
```

Comment:
I haven't done an actual test. Will the error raised multiple times? do you want to make it
```
elsif task.get_option('cancel_requested') && task.get_option(:cleanup_request_id).absent?
```

again here we have a mix use of symbol and string. `'cancel_requested'` should be `:cancel_requested`. When I created the PR I did not pay attention to what existing keys are. We can leave it as is since this is a temporary solution.

Reply:
Yes, it will call `create_cleanup_request` multiple times, but it will do nothing and return nil.It doesn't seem to add overload. I'd say "as you wish" ;)